src/run/runner/wall_time/perf.rs

```diff
@@ -0,0 +1,84 @@
+use crate::run::runner::helpers::setup::run_with_sudo;
+use codspeed::fifo::FifoIpc;
+use std::io::{Read, Write};
+
+pub const PERF_CTL_FIFO: &str = "/tmp/codspeed_perf.ctl.fifo";
+pub const PERF_CTL_ACK_FIFO: &str = "/tmp/codspeed_perf.ack.fifo";
+
+pub struct PerfFifo {
+    ctl_fifo: FifoIpc,
+    ack_fifo: FifoIpc,
+}
+
+impl PerfFifo {
+    fn wait_for_ack(&mut self) {
```

Comment:
This could be an  async function to avoid polling, and the use case matches well the purpose of the event loop